core/src/main/scala/io/finch/request/package.scala

```diff
@@ -34,6 +34,55 @@ import scala.reflect.ClassTag
 package object request {
 
   /**
+   * A reusable validation rule that can be applied to any ''RequestReader'' with a matching type.
+   */
+  class ValidationRule[T] private[request] (val description: String, predicate: T => Boolean) extends (T => Boolean) { self =>
```

Comment:
What do you think about implementing it as 
```scala
trait ValidationRule[A] { 
  def description: String
  def apply(value: A): Boolean
  ...
}
```

And then:

```scala
object ValidationRule {
  def apply[A](d: String)(p: A => Boolean): ValidationRule[A] = new ValidationRule[A] {
    def description = d
    def apply(value: A) = p(value)
  }
}
```

I know that this is just implementation details but this approach fits well in to the whole system, since other primitives (i.e., `RequestReader` uses it). Also could you please use `A` instead of `T`?

Sorry for being picky :) I just looked through the PR and found out that we may make it look like other Finch code.